### PR TITLE
Unlock IP Search for users

### PR DESCRIPTION
**Pull request by Dane2007**
Now identified to Wikimedia Foundation.  This should be unlocked for certain research purposes.

### CODE REVIEW

File: templates/search/searchform.tpl [hunk 1]
```diff
@@ -11,11 +11,11 @@
       <select name="type">
         <option value="Request">... requested username</option>
         <option value="email">... email address</option>
+				<option value="IP">... IP address</option>
```
Comment (stwalkerster) [inline templates/search/searchform.tpl]: You might want to fix the indentation here

File: templates/search/searchform.tpl [hunk 2]
```diff
@@ -11,11 +11,11 @@
...
         {if $currentUser->isAdmin() || $currentUser->isCheckuser()}
```
Comment (stwalkerster) [inline templates/search/searchform.tpl]: ... and while you're editing this file, let's remove the useless CIDR options and the duplicate IP option if you're a tool admin...

... which leaves the entire if block empty, and thus also unnecessary.

File: templates/search/searchform.tpl [hunk 3]
```diff
@@ -11,12 +11,9 @@
...
         <option value="Request">... requested username</option>
         <option value="email">... email address</option>
+	<option value="IP">... IP address</option>
         {if $currentUser->isAdmin() || $currentUser->isCheckuser()}
         <option value="IP">... IP address</option>
-        <option value="CIDR" disabled="disabled">... CIDR Range</option>{* not implemented yet! *}
-        {else}
-        <option value="IP" disabled="disabled">... IP address</option>
-        <option value="CIDR" disabled="disabled">... CIDR Range</option>
         {/if}
```
Comment (FastLizard4) [inline templates/search/searchform.tpl]: Lines 15 through 17 are now completely redundant and should be removed entirely; as is, the code will show the IP Address option to admins and checkusers twice.

File: search.php [hunk 4]
```diff
@@ -100,14 +100,6 @@
...
 	}
 	elseif ($_GET['type'] == 'IP') {
-		// move this to here, so non-admins can perform searches, but not on IP addresses or emails
-		if (!User::getCurrent()->isAdmin() && !User::getCurrent()->isCheckuser()) {
-			// Displays both the error message and the footer of the interface.
-			BootstrapSkin::displayAlertBox("IP address search is only available to tool admins and checkusers.", "alert-error", "Access Denied");
-			$smarty->display("search/searchform.tpl");
-			BootstrapSkin::displayInternalFooter();
-			die();
-		}
```
Comment (FastLizard4) [inline search.php]: Code style pedant: Remove this blank line

**Pull request merged**